providers/base/bin/network.py

```diff
@@ -288,7 +288,7 @@ def run(self):
         t = []
         results.clear()
         for thread_num in range(0, python_threads):
-            if self.iperf3:
+            if self.iperf3 and len(core_list) > 0:
```

Comment:
Note: Checking if len(core_list) > 0 is essentially the same as checking if core_list. In Python, empty collections are considered False when converted to a boolean context, so you can simplify your condition check. Of course it's an extremely minor thing, and the PR is great as is!